lib/puppet/indirector/hiera.rb

```diff
@@ -18,6 +18,7 @@ def initialize(*args)
   def find(request)
     not_found = Object.new
     options = request.options
+    Puppet.debug "Performing a hiera indirector lookup of #{request.key} with options #{options.inspect}"
```

Comment:
Always call `Puppet.debug` with a no-args-lambda if the argument requires evaluation (here it requires interpolation). i.e. 
```
Puppet.debug { "Performing a hiera indirector lookup of #{request.key} with options #{options.inspect}" }
```